etc/systemd/dnf-system-upgrade.service

```diff
@@ -1,14 +1,16 @@
 [Unit]
-Description=System Upgrade
+Description=System Upgrade using DNF
 ConditionPathExists=/system-update/.dnf-system-upgrade
 Documentation=http://www.freedesktop.org/wiki/Software/systemd/SystemUpdates
 
 DefaultDependencies=no
 Requires=sysinit.target
 After=sysinit.target systemd-journald.socket
-Before=shutdown.target
+Before=shutdown.target system-update.target
```

Comment:
Are you sure this is correct? This unit gets installed into the `system-update.target` target anyway, so I'm not sure this makes sense.